src/lib/staged_ledger/staged_ledger_diff.ml

```diff
@@ -289,4 +303,22 @@ end) :
   let user_commands (t : t) =
     (fst t.diff).user_commands
     @ Option.value_map (snd t.diff) ~default:[] ~f:(fun d -> d.user_commands)
+
+  let completed_works (t : t) =
+    (fst t.diff).completed_works
+    @ Option.value_map (snd t.diff) ~default:[] ~f:(fun d -> d.completed_works)
+
+  let add_coinbase (coinbases : Ft.t list) =
+    List.fold coinbases ~init:Currency.Fee.zero
+      ~f:(fun acc (_, fee_transfer) ->
+        Option.value_exn (Currency.Fee.add acc fee_transfer) )
+
+  let total_coinbase (t : t) =
+    let first_pre_diff, second_pre_diff_opt = t.diff in
+    let fee_transfers =
+      At_most_two.to_list first_pre_diff.coinbase
+      @ Option.value_map second_pre_diff_opt ~default:[]
```

Comment:
`fee_transfer` here is the snark fee that is paid to the snark worker which is part of the snark-fee that you're already calculating. What you want is something like this: 
```ocaml 
match first_pre_diff.coinbase,  Option.value_map second_pre_diff_opt ~default:Zero ~f:(fun d -> d.coinbase) with 
| At_most_two.Zero, At_most_one.Zero -> Currency.Amount.zero
| _ -> Coda_praos.coinbase_amount
```

But if `Coda_praos.coinbase_amount` changes then it could show incorrect reward. We might need the rewarded `coinbase_amount` in the diff in that case. @bkase  /@imeckler any thoughts?